File: haikuReview/20140213-xavi-someone/src/main/java/Haiku.java
import java.util.Arrays;
import java.util.List;
import java.util.regex.Pattern;

public class Haiku {

    private final String haiku;

    private Pattern linesPattern = Pattern.compile("/");

    public Haiku(String haiku) {
        this.haiku = haiku;
    }

    boolean isValidLineCount() {
        return haiku.matches("(.*)/(.*)/(.*)");
    }

    public List<String> asLines() {
        String[] split = linesPattern.split(haiku);
        List<String> strings = Arrays.asList(split);
        if (!isValidLineCount()) throw new RuntimeException("A haiku yoyo, it should be 3 lines only, try again carrot");
        return strings;
    }


}

File: haikuReview/20140213-xavi-someone/src/main/kotlin/Haiku.kt
/**
 * Created by xrigau on 13/02/2014.
 */


import java.util.Arrays

public open class Haiku2(haiku : String) {

    private val haiku : String

    open fun isValidLineCount() : Boolean {
        return haiku.matches("(.*)/(.*)/(.*)")!!
    }

    public open fun asLines() : java.util.List<String> {
        val split = haiku.split("/")
        val strings = Arrays.asList(split) as java.util.List<String>
        if (isValidLineCount())
            throw RuntimeException("A haiku yoyo, it should be 3 lines only, try again carrot")
        return strings
    }

    {
        this.haiku = haiku
    }
}
